### PR TITLE
Widget with unread Badge

### PR DESCRIPTION
**Pull request by zoff99**

### CODE REVIEW

**Comment by zoff99** [comment]
instead of puttin the normal app-icon on your homescreen, just put this new widget on your homescreen.
it does the same as the normal icon but displays the unread count on the icon

**Comment by zoff99** [comment]
This is what it looks like

Normal App Icon on the left // UnreadWidget on the right

**Comment by pR0Ps** [comment]
I like the idea here since it makes unread bubbles available to all users (unlike other methods which require a custom launcher) while keeping it entirely optional.

Would there be a nicer way of integrating your [standalone repo for this](https://github.com/zoff99/UnreadWidget) vs. just copy/pasting the code in?

**Comment by zoff99** [comment]
you can integrate it however you like.

**Comment by p4nci** [comment]
@zoff99 is there any reason why in the widget the orig Silence icon was resized (smaller)?

silence-widget

**Comment by zoff99** [comment]
it is not on purpose. widgets do not render the same as the launcher icons.
maybe you can fix it?
but be careful widgets sizes are tricky on older phones (android 2.3)

**Comment by zoff99** [comment]
i just wonder will someone finally merge this? or is there still some discussion?
a status would be nice.

**Comment by BastienLQ** [comment]
@Zoff We will merge this PR if you move this functionality into a
library or if you improve your code to match Silence's Code Style
Guidelines and Silence's package name. Actually, I could do the second
option, but I'm focusing on #390 right now.

**Comment by zoff99** [comment]
the problem is i don't know how to make it a library and also if that is even possible, since it needs some values from the project. so this option is out, i am afraid.

not sure why it needs to be a library? source is source, no?

File: src/com/zoffcc/applications/CustomWidgetProvider.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+/**
+ * Created on 01.09.2016.
+ */
+public class CustomWidgetProvider extends AppWidgetProvider
+{
+
+    // ---- CONFIG ----
+    // ---- CONFIG ----
+    // ---- CONFIG ----
+    public static final int MAX_COUNT = 20;
```
Comment (p4nci) [inline src/com/zoffcc/applications/CustomWidgetProvider.java]: suggestion: set MAX_COUNT value at least to 99, it's still 2-digit number (better 999 if possible)

**Comment by BastienLQ** [comment]
Closing in favour of #511.

**Pull request closed without merging**